Modules/Loadable/Markups/MRML/vtkMRMLMarkupsCurveNode.cxx

```diff
@@ -183,6 +194,182 @@ double vtkMRMLMarkupsCurveNode::GetCurveLengthBetweenStartEndPointsWorld(vtkIdTy
   }
 }
 
+//---------------------------------------------------------------------------
+void vtkMRMLMarkupsCurveNode::ResampleCurveSurface(double controlPointDistance, vtkMRMLModelNode* modelNode, double maximumSearchRadius)
+{
+	if(!modelNode)
+	  {
+	  vtkErrorMacro("vtkMRMLMarkupsCurveNode::ResampleCurveSurface failed: Constraint surface is not valid");
+	  }
+	vtkPoints* originalPoints = this->GetCurvePointsWorld();
+	if (!originalPoints  || originalPoints->GetNumberOfPoints() < 2)
+      {
+	  return;
+      }
+	vtkNew<vtkPoints> originalControlPoints;
+	this->GetControlPointPositionsWorld(originalControlPoints);
+	vtkNew<vtkStringArray> originalLabels;
+	this->GetControlPointLabels(originalLabels);
+
+	vtkNew<vtkPoints> interpolatedPoints;
+	vtkMRMLMarkupsCurveNode::ResamplePoints(originalPoints, interpolatedPoints, controlPointDistance, this->CurveClosed);
+
+	vtkPolyData* surfacePolydata = modelNode->GetPolyData();
+	if(!surfacePolydata)
+	  {
+	  vtkErrorMacro("vtkMRMLMarkupsCurveNode::ResampleCurveSurface failed: Constraint surface polydata is not valid");
+	  }
+	vtkNew<vtkPointLocator> pointLocator;
+	pointLocator->SetDataSet(surfacePolydata);
+	pointLocator->BuildLocator();
+
+	vtkDataArray* normalVectorArray = vtkArrayDownCast<vtkDataArray>(surfacePolydata->GetPointData()->GetArray("Normals"));
+	if(!normalVectorArray)
+	  {
+	  vtkNew<vtkPolyDataNormals> normalFilter;
+	  normalFilter->SetInputData(surfacePolydata);
+	  normalFilter->ComputePointNormalsOn();
+	  normalFilter->Update();
+	  vtkPolyData* normalPolydata = normalFilter->GetOutput();
+	  vtkDataArray* normalVectorArray = vtkArrayDownCast<vtkDataArray>(normalPolydata->GetPointData()->GetNormals());
+	  if (!normalVectorArray)
+	    {
+		vtkGenericWarningMacro("vtkMRMLMarkupsCurveNode::ResamplePoints failed: Unable to find or calculate normals");
```

Comment:
Generic warning macro is only intended for static functions, we should use vtkWarningMacro here. But since the operation fails, it would be better to log an error instead.